src/frame.rs

```diff
@@ -204,11 +206,18 @@ impl Frame {
         let mut data_vec = Vec::with_capacity(num_planes);
         let linesize = self.frame.Linesize;
 
+        // very sketchy, try to find a better way
+        let pix_fmt: AVPixelFormat =
+            unsafe { mem::transmute(self.frame.EncodedPixelFormat) };
+        let pix_fmt = Pixel::from(pix_fmt);
+        let log2_chroma_h = pix_fmt.descriptor().unwrap().log2_chroma_h();
+
         for i in 0..num_planes {
             if linesize[i] == 0 {
                 data_vec.push(None);
             } else {
-                let plane_slice_length = linesize[i] * frame_resolution.height;
+                let plane_slice_length = linesize[i] * frame_resolution.height
+                    / if i == 0 { 1 } else { 1 << log2_chroma_h };
```

Comment:
We can use `EncodedHeight` as [here](https://github.com/FFMS/ffms2/blob/master/test/tests.cpp#L38) and let's put the `if` in another variable like the [example](https://github.com/FFMS/ffms2/blob/master/test/tests.cpp#L36)

Reply:
We should copy the same code from the example in my opinion, so let's use right shifts instead of divisions